src/test/java/plugins/MSBuildPluginTest.java

```diff
@@ -0,0 +1,133 @@
+package plugins;
+
+import static org.junit.Assert.assertThat;
+
+import java.io.IOException;
+
+import org.apache.commons.lang3.SystemUtils;
+import org.hamcrest.Matchers;
+import org.jenkinsci.test.acceptance.junit.AbstractJUnitTest;
+import org.jenkinsci.test.acceptance.junit.WithPlugins;
+import org.jenkinsci.test.acceptance.msbuild.MSBuildInstallation;
+import org.jenkinsci.test.acceptance.msbuild.MSBuildStep;
+import org.jenkinsci.test.acceptance.po.Build;
+import org.jenkinsci.test.acceptance.po.FreeStyleJob;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.internal.AssumptionViolatedException;
+
+/**
+ * This test is designed to be run on a Windows machine with MSBuild installed
+ * 
+ * In order for the test to know where the MSBuild installation dir is, an
+ * environment variable must be set. i.e:
+ * 
+ *      MSBUILD_EXE = C:\Windows\Microsoft.NET\Framework\v2.0.50727\MSBuild.exe
+ * 
+ * Tests will be skipped if:
+ *      - The environment variable is not set 
+ *      - The test is not running on a Windows machine
+ *      - MSBUild.exe is not present in the specified location
+ */
+@WithPlugins("msbuild")
+public class MSBuildPluginTest extends AbstractJUnitTest {
+
+    private static final String MSBUILD_EXE_ENV = "MSBUILD_EXE";
+    private static final String MSBUILD_NAME = "MSBuildInstallation";
+
+    /**
+     * Builds a FreeStyle job with a MSBuild step with the configuration passed as parameter
+     * 
+     * @param workspacePath The workspace where the project is located
+     * @param buildFile The build file (.proj or .sln)
+     * @param cmdArguments the command line arguments to execute. Can be null.
+     * @return a job
+     * @throws IllegalArgumentException if workspacePath or buildFile are null
+     */
+    private FreeStyleJob msBuildJob(String workspacePath, String buildFile, String cmdArguments) {
+        if (workspacePath != null && buildFile != null) {
+            FreeStyleJob job = jenkins.jobs.create(FreeStyleJob.class);
+            job.copyDir(resource(workspacePath));
+            MSBuildStep msBuildStep = job.addBuildStep(MSBuildStep.class);
+            msBuildStep.setMSBuildName(MSBUILD_NAME).setMSBuildFile(buildFile);
+            if (cmdArguments != null && !cmdArguments.isEmpty()) {
+                msBuildStep.setCmdLineArgs(cmdArguments);
+            }
+            job.save();
+            return job;
+        } else {
+            throw new IllegalArgumentException("Workspace and buildFile must be different from null.");
+        }
+    }
+    
+    @Before
+    public void setUp() {
+         //Check environment configuration
+         if (System.getenv(MSBUILD_EXE_ENV) == null) {
```

Comment:
I am afraid I had already tried that and it does not work. I wonder why this is being restricted this way. It just makes it more difficult to run tests and can cause problems depending on the CI machine that runs them. 

Currently, all tests using `@TestActivation`are being ignored on the CI server. See [1](https://jenkins.ci.cloudbees.com/job/core/job/acceptance-test-harness/1630/testReport/plugins/BeakerBuilderPluginTest/), [2](https://jenkins.ci.cloudbees.com/job/core/job/acceptance-test-harness/1630/testReport/plugins/OpenstackCloudPluginTest/), [3](https://jenkins.ci.cloudbees.com/job/core/job/acceptance-test-harness/1630/testReport/plugins.jclouds/JCloudsPluginTest/) and [4](https://jenkins.ci.cloudbees.com/job/core/job/acceptance-test-harness/1630/testReport/core/JdkTest/). But this is probably because the needed [environment variables](https://jenkins.ci.cloudbees.com/job/core/job/acceptance-test-harness/1630/injectedEnvVars/) are not set. What I don't know is whether they are not set because nobody did it, or because nobody was able to.

I think it would make sense to change the `.` for a `_` and I am happy to do it. I would also update the only 4 plugin tests that use this annotation. What do you think?